analyses/npl/03_flood_events.md

```diff
@@ -0,0 +1,327 @@
+                (df_events_high_impact['basin'] == basin) & (df_events_high_impact['impact_parameter'] == impact_parameter)
+            ]
+            df_events_sub['detections'] = 0
+            TP = 0
+            FP = 0
+            rp_val = df_return_period.loc[rp, station]
+            observations = ds_glofas_reanalysis[station].values
+            groups = utils.get_groups_above_threshold(observations, rp_val, min_duration=ndays)
+            for group in groups:
+                # The GlofAS event takes place on the Nth day (since for an event)
+                # you require N days in a row
+                glofas_event_date = ds_glofas_reanalysis.time[group[0] + ndays - 1].data
+                # Check if any events are around that date
+                days_offset = (df_events_sub['date'] - glofas_event_date) /  np.timedelta64(1, 'D')
+                detected = (days_offset > -1 * days_before_buffer) & (days_offset < days_after_buffer)
+                df_events_sub.loc[detected, 'detections'] += 1
+                # If there were any detections, it's  a TP. Otherwise a FP
+                if sum(detected):
+                    TP += 1
+                else:
+                    FP += 1
+            df_station_stats = df_station_stats.append({
+                'station': station,
+                'impact_parameter': impact_parameter,
+                'TP': TP,
+                'FP': FP,
+                'FN': len(df_events_sub[df_events_sub['detections'] == 0])
+            }, ignore_index=True)
+```
+
+### Plot events and GloFAS together
+
+```python
+basin = 'Koshi'
+impact_parameter = 'Total Death'
+df_events_sub = df_events_high_impact[
+                (df_events_high_impact['basin'] == basin) & (df_events_high_impact['impact_parameter'] == impact_parameter)
+            ]
+df_events_sub
+```
+
+```python
+rp = 5
+for basin, station_list in STATIONS.items():
+    for station in station_list:
+        fig, axs = plt.subplots(len(impact_parameters), figsize=(12, 10))
+        fig.suptitle(f'{basin} - {station}')
+        fig.supylabel('Discharge [m$^3$ s$^{-1}$]')
+        for i, impact_parameter in enumerate(impact_parameters):
+            df_events_sub = df_events_high_impact[
+                (df_events_high_impact['basin'] == basin) & (df_events_high_impact['impact_parameter'] == impact_parameter)
+            ]
+            observations = ds_glofas_reanalysis[station].values
+            x = ds_glofas_reanalysis.time
+            ax = axs[i]
+            ax.plot(x, observations, lw=0.5)
+            # Plot when GLoFAS is above RP
+            rp_val=df_return_period.loc[rp, station]
+            groups = utils.get_groups_above_threshold(observations, rp_val, ndays)
+            for group in groups:
+                idx = range(group[0], group[1])
+                ax.plot(x[idx], observations[idx], ls='-', 
+                        lw=2, c='C1')
+            ax.axhline(y=rp_val, c='C1')
```

Comment:
Maybe this is added in the slides, but could add a legend here to indicate the rp level for more clarity.

Reply:
Good point, done!